overlap-scripts/kubectl-kubestellar-ensure-mbw

```diff
@@ -0,0 +1,152 @@
+#!/usr/bin/env bash
+
+# Copyright 2023 The KubeStellar Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Purpose: ensure that a mailbox workspace has the needed KubeStellar CRDs.
+
+# Usage: $0 ($kubectl_flag | --with-kube boolean | -X)* mbw_name
+
+want_kube=true
+mbw_name=""
+
+while (( $# > 0 )); do
+    case "$1" in
+	(-h|--help)
+	    echo "Usage: kubectl kubestellar ensure mbw (\$kubectl_flag | --with-kube boolean | -X)* wm_workspace_name"
+	    exit 0;;
+	(-X) set -o xtrace;;
+	(--with-kube)
+	    if (( $# >1 ))
+	    then want_kube="$2"; shift
+	    else echo "$0: missing with-kube value" >&2; exit 1
+	    fi;;
+	(--context*)
+	    # TODO: support --context
+	    echo "$0: --context flag not supported" >&2; exit 1;;
+	(--*=*|-?=*)
+	    kubectl_flags[${#kubectl_flags[*]}]="$1";;
+	(--*|-?)
+	    kubectl_flags[${#kubectl_flags[*]}]="$1";
+	    if (( $# > 1 )); then 
+		 kubectl_flags[${#kubectl_flags[*]}]="$2"
+		 shift
+	    fi;;
+	(-*)
+	    echo "$0: flag syntax error" >&2
+	    exit 1;;
+	(*)
+	    if [ -z "$mbw_name" ]
+	    then mbw_name="$1"
+	    else echo "$0: too many positional arguments" >&2
+		 exit 1
+	    fi
+    esac
+    shift
+done
+
+if [ "$mbw_name" == "" ]; then
+    echo "$0: workload management workspace name not specified" >&2
+    exit 1
+fi
+
+case "$want_kube" in
+    (true|false) ;;
+    (*) echo "$0: with-kube should be true or false" >&2
+	exit 1;;
+esac
+
+set -e
+
+kubectl ws "${kubectl_flags[@]}" root
```

Comment:
This side-effects the kubeconfig file. What prevents concurrent reads and writes to that file?

Reply:
This has been addressed in #1301